solver.go

```diff
@@ -361,34 +361,43 @@ func (s *solver) solve() (map[atom]map[string]struct{}, error) {
 		// that has yet to be selected.)
 		if awp, is := s.sel.selected(bmi.id); !is {
 			s.mtr.push("new-atom")
-			// Analysis path for when we haven't selected the project yet - need
-			// to create a version queue.
-			queue, err := s.createVersionQueue(bmi)
-			if err != nil {
-				// Err means a failure somewhere down the line; try backtracking.
-				s.traceStartBacktrack(bmi, err, false)
+
+			if !bmi.fromRoot {
+				// Here we have a flat package, and we don't need to check any dependencies
+				// for it. There is nothing to check here, so remove it.
+				s.unsel.remove(bmi)
```

Comment:
This isn't doing what you want it to :)

First, if the bmi does represent a "flat package", then that doesn't guarantee anything about whether or not the package has external imports. This point in the algorithm is along the "unselected" path (`!is`), meaning that it's the first time we're seeing a bmi for this particular `ProjectRoot`. We have to follow the existing logic, because that's what will look at all of its imports, pull in its manifest, and do the satisfiability checks.

If we were on the other branch - `is` - then this sort of an escape hatch would make sense. We don't need it, though, because...

Hmm. Well, I'd say that it's because, in `selectAtom()`, we only add bmis to the unselected queue if the package set we need is unvisited. But the check there (https://github.com/sdboyer/gps/blob/master/solver.go#L1113-L1124) doesn't seem like quite what I'd expect. Let me refresh myself on it...